jetty-core/jetty-core-server/src/test/java/org/eclipse/jetty/core/server/ServerConnectorHttpServerTest.java

```diff
@@ -24,6 +34,88 @@
     public void init() throws Exception
     {
         // Run this test with 0 acceptors. Other tests already check the acceptors >0
-        startServer(new ServerConnector(_server, 0, 1));
+        initServer(new ServerConnector(_server, 0, 1));
+    }
+
+    @Test
+    public void testNonBlockingInvocationType() throws Exception
+    {
+        startServer(new Handler.Abstract()
+        {
+            @Override
+            public void handle(Request request) throws Exception
+            {
+                request.accept().getCallback().succeeded();
+            }
+
+            @Override
+            public InvocationType getInvocationType()
+            {
+                return InvocationType.NON_BLOCKING;
+            }
+        });
+
+        try (Socket client = newSocket(_serverURI.getHost(), _serverURI.getPort()))
+        {
+            OutputStream os = client.getOutputStream();
+
+            String request = """
+                GET / HTTP/1.1\r
+                Host: localhost\r
+                \r
+                """;
+            os.write(request.getBytes(StandardCharsets.ISO_8859_1));
+            os.flush();
+
+            // Read the response.
+            String response = readResponseHeader(client);
+
+            assertThat(response, containsString("HTTP/1.1 200 OK"));
+        }
+
+        // TODO need a better way to test this
+        assertThat(_connector.getSelectorManager().getBean(ManagedSelector.class).dump(), not(containsString("[pc=0")));
+        assertThat(_connector.getSelectorManager().getBean(ManagedSelector.class).dump(), containsString("pic=0,pec=0,epc=0]"));
```

Comment:
@sbordet Can you think of a better way to test this?

Reply:
Ensure there is only 1 selector when configuring the connector, then:

```java
ManagedSelector selector = _connector.getSelectorManager().getBean(ManagedSelector.class);
AdaptiveExecutionStrategy strategy = selector.getBean(AdaptiveExecutionStrategy.class);
assertEquals(0, strategy.getPICTasksExecuted());
```